app/services/user_signup.rb

```diff
@@ -0,0 +1,52 @@
+class UserSignup
+
+  attr_reader :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+          amount: 999, # amount in cents, again
+          source: stripe_token,
+          description: "Sign up charge for #{@user.email}"
+        )
+      if charge.successful? 
+        @user.save
+        handle_invitation(invitation_token)
+        AppMailer.delay.send_welcome_email(@user)
+        @status = :success
+        self
+      else
+        @status = :failed
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :failed
+      @error_message = "Invalid user information. Please check the errors below."
+      self
```

Comment:
You can refactor each `self` to the bottom of this method. Each `if` branch ends up returning it.